### ISSUE TITLE
Validation webhook must be created, even if create is disabled in values.yaml

### ISSUE THREAD

**Issue by feenr** [cloudnative-pg/cloudnative-pg#2518]
I configured cnpg to skip creation of mutation and validation webhooks because I wanted to provide my own, however, when the operator was starting up it was getting an error because the validating webhook was not present on the system. This behavior seems to conflict with the configuration settings.

To get around this problem I set WebhookCertDir in the cnpg config map, which is not a documented value and isn't really intended for this exact purpose. Please consider skipping webhook validation when their creation is disabled.

[https://github.com/cloudnative-pg/cloudnative-pg/blob/c59dbabadce6471054c5c58689efe437c261ae2b/internal/cmd/manager/controller/controller.go#L3[…]71](https://github.com/cloudnative-pg/cloudnative-pg/blob/c59dbabadce6471054c5c58689efe437c261ae2b/internal/cmd/manager/controller/controller.go#L345-L371)

**Comment by phisco** [comment]
🤔 @feenr I was thinking about this. So, the chart allows not to create webhooks, but the operator then looks for them and fails, which is indeed something we should align. 

Right now we support the following scenarios:
- usual setup: webhooks enabled from the chart, webhooks created, webhook server running, certs handled by the operator.
- bring your own certs: webhooks enabled from the chart, webhooks created, webhook server running, certs not handled by the operator but provided via WebhookCertDir.

If I got it correctly, what you are asking for is to also support the following:
- no webhooks at all: webhooks disabled from the chart, no webhooks created, no webhook server running, no certs to be handled

Is this right?

**Comment by feenr** [comment]
You've got it right. We have an alternative kind that supports webhooks and
I've been able to get things working with them.

**Comment by phisco** [comment]
Then you still need the webhook server to be running, just not the validating webhook to be created, nor it's CA to be handled, right? Hence you need some kind of "skip certs handling" flag. Otherwise what would be answering your webhook calls?

You mentioned you set webhookcertdir to make it work. What did you set it to? Because most probably your own kind of webhooks still needs the webhook's cabundle to match the server certificate used by the webhook server, no? So actually setting webhook cert dir was what you needed and we should just document that.

**Comment by feenr** [comment]
Yes. I have the mutation and validation web hooks both running, but defined
in a separate resource which handles certificates. I'm setting the webhook
cert dir in cn-pg to an arbitrary value, ie /temp.

What I am hoping for is a new configuration in the manager image which
skips webhook pk validation. The helm chart could set the configuration
based upon whether webhook creation is skipped. (Not sure exactly how this
would work since the helm chart isn't creating a configmap currently I
believe)

If my use case is too niche I'd understand and settle with my current
approach.

Thanks Phillipe!

**Comment by feenr** [comment]
@phisco I discovered today that this is behaving worse than I realized. When I got the cluster up an running I still had webhook.validating.create set to true. If I kept that as false, even with `WebhookCertDir` I'd get a validation error at another point in manager startup. 

https://github.com/cloudnative-pg/cloudnative-pg/blob/main/internal/cmd/manager/controller/controller.go#L281

The error says unable to find file "/run/secrets/cnpg.io/webhook/tls.crt" which means that `WebhookCertDir` is being ignored, as it's value has not been loaded until a later step:
https://github.com/cloudnative-pg/cloudnative-pg/blob/main/internal/cmd/manager/controller/controller.go#L157
https://github.com/cloudnative-pg/cloudnative-pg/blob/main/internal/cmd/manager/controller/controller.go#L176

So although I can get cn-pg working with my environmental constraints it feels a bit like a house of cards the way I have it set up.

**Comment by Stevenpc3** [comment]
I have a need to fully disable webhooks as we do not need them.  At the very least we would like to create the items as "namespaced" items.  We require deploying an operator PER namespace to ONLY monitor it's own namespace.  We have updated the chart to add the env which CNPG supports to --watch-namespace, but when the chart is deployed it creates these items with no namespace and worse the hook only monitors it's own namspace.  Now we cannot deploy another chart because this item already exists and helm fails to do so.  And we cannot disable the hook...

**Comment by Stevenpc3** [comment]
Nevermind... using the create false for muting and validating works and hooks are not created.

https://github.com/cloudnative-pg/charts/blob/main/charts/cloudnative-pg/values.yaml#L40

```
webhook:
  mutating:
    create: false
  validating:
    create: false
```
I assume this issue is no longer valid.
